pyspcwebgw/__init__.py

```diff
@@ -99,23 +99,35 @@ async def _async_ws_handler(self, data):
         _LOGGER.debug("SIA code is %s for ID %s", sia_code, spc_id)
 
         if sia_code in Area.SUPPORTED_SIA_CODES:
-            entity = self._areas.get(spc_id, None)
+            entities = [self._areas.get(spc_id, None)]
             resource = 'area'
         elif sia_code in Zone.SUPPORTED_SIA_CODES:
-            entity = self._zones.get(spc_id, None)
+            entities = [self._zones.get(spc_id, None)]
             resource = 'zone'
+        elif sia_code in ('CL', 'OP'):
+            # workaround for area mode change update in certain firmwares
+            # sia_code is in fact the user performing the change so
+            # refresh all areas
```

Comment:
I don't think this is a firmware thing, but a setting.
This is needed if you only use one area, and then disable the use of different areas in the SPC config settings.

Reply:
...but maybe it is better to let it be like this for now, and make small cosmetic changes later.